src/platform/plugins/shared/ai_assistant_management/selection/server/plugin.ts

```diff
@@ -152,9 +110,46 @@ export class AIAssistantManagementSelectionPlugin
       },
     });
 
+    this.registerUiSettings(core, plugins);
+
     return {};
   }
 
+  private registerUiSettings(
+    core: CoreSetup,
+    plugins: AIAssistantManagementSelectionPluginServerDependenciesSetup
+  ) {
+    const { cloud } = plugins;
+    const serverlessProjectType = cloud?.serverless.projectType;
+
+    switch (serverlessProjectType) {
+      case 'observability':
+        core.uiSettings.register({
+          [PREFERRED_AI_ASSISTANT_TYPE_SETTING_KEY]: {
+            ...observabilitySolutionSetting,
+            value: this.config.preferredAIAssistantType,
+          },
+        });
+        return;
+      case 'security':
+        core.uiSettings.register({
+          [PREFERRED_AI_ASSISTANT_TYPE_SETTING_KEY]: {
+            ...securitySolutionSetting,
+            value: this.config.preferredAIAssistantType,
+          },
+        });
+        return;
+      default:
+        // This case is hit when in stateful Kibana
```

Comment:
can you put a TODO here somewhere that this also needs to be addressed for Search, if we don't do it in this PR?